src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -3004,6 +3183,54 @@ AggregateFunctionOid(const char *functionName, Oid inputType)
 }
 
 
+/*
+ * AggregateFunctionOid performs a reverse lookup on aggregate function name,
+ * and returns the corresponding aggregate function oid for the given function
+ * name and input type.
+ */
+static Oid
+AggregateFunctionOidWithoutInput(const char *functionName)
+{
+	Oid functionOid = InvalidOid;
+	Relation procRelation = NULL;
+	SysScanDesc scanDescriptor = NULL;
+	ScanKeyData scanKey[1];
+	int scanKeyCount = 1;
+	HeapTuple heapTuple = NULL;
+
+	procRelation = heap_open(ProcedureRelationId, AccessShareLock);
+
+	ScanKeyInit(&scanKey[0], Anum_pg_proc_proname,
+				BTEqualStrategyNumber, F_NAMEEQ, CStringGetDatum(functionName));
+
+	scanDescriptor = systable_beginscan(procRelation,
+										ProcedureNameArgsNspIndexId, true,
+										NULL, scanKeyCount, scanKey);
+
+	/* loop until we find the right function */
```

Comment:
where is the "loop"?

Reply:
This whole function was thrown together too hastily; it'd easily grab the wrong thing if a user created functions with colliding names. Updated to lookup with qualified name & input types